### DISCUSSION TITLE
Adding new language for code highlighting

### DISCUSSION THREAD

**Discussion by jeepers3327** [facebook/lexical#2176] [Q&A]
Looking at `@lexical/code` package, it has fixed sets of prismjs highlight styles.
```
import 'prismjs/components/prism-clike';
import 'prismjs/components/prism-javascript';
import 'prismjs/components/prism-markup';
import 'prismjs/components/prism-markdown';
import 'prismjs/components/prism-c';
import 'prismjs/components/prism-css';
import 'prismjs/components/prism-objectivec';
import 'prismjs/components/prism-sql';
import 'prismjs/components/prism-python';
import 'prismjs/components/prism-rust';
import 'prismjs/components/prism-swift';
```

Is there a way to add  or extend the package to add additional prismjs package to support other programming languages?

**Comment by fantactuka** [comment]
Prism adds itself globally (into window), so you should be able to import any prism language definition and it'll be available to lexical. E.g. if you want to add Lua support:

```
import MyLexicalEditor from './Editor';
import 'prismjs/components/prism-lua';
```

**Reply by jLynx** [reply]
Do you know how we can load all languages?
